generate_manifest.R

```diff
@@ -0,0 +1,31 @@
+#############
+# Get a manifest of files to upload
+#############
+
+#############
+### Synapse credentials and params
+#############
+# Set the environment variables .Renviron file in your home folder. Refer to README for more details
+SYNAPSE_USERNAME= Sys.getenv('SYNAPSE_USERNAME')
+SYNAPSE_PASSWORD= Sys.getenv('SYNAPSE_PASSWORD')
```

Comment:
I would use a synapse personal access token for security purposes

This also simplifies your code by quite a bit because by setting this env variable:

`export SYNAPSE_AUTH_TOKEN=...`

Both the python and R client will read from this and login without you having to specify the login.